### ISSUE TITLE
Replay Debugger doesn't work when Logged in As different user

### ISSUE THREAD

**Issue by suneelsfdc** [forcedotcom/salesforcedx-vscode#1280]
### Summary

I have tried using Apex Replay Debugger by logging in as a different user(Going to a user record and hitting Login As). User U1 logged into Salesforce, later he/she opened another user record U2 and logged in as him/her and performed some action, but replay debugger doesn't generate logs

### Steps To Reproduce:

1.  Setup a Salesforce VS Code project and install Salesforce Extension for VS Code
2.  Goto Debug screen and select Apex Launch Replay Debugger configuration
2.  Launch Salesforce instance as the same user as VS Code project creator
3.  Turn on Apex Replay Debugger from VS Code
4.  Go to a user record and login as the user
5.  Do some action that can generate the log
6.  Turn off Apex Replay Debugger from VS Code
7.  Get the logs from VS Code

### Expected result

Debugger should have generated logs for the newly logged in user(Login As user)

### Actual result

Debugger didn't generate any new logs

### Additional information

_Feel free to attach a screenshot_.

**VS Code Version**:

**SFDX CLI Version**:

**OS and version**:

**Comment by lcampos** [comment]
This is not a bug but a misconception on how the Apex Replay Debugger works. When you turn on Apex Replay Debugger (`SFDX: Turn On Apex Debug Log for Replay Debugger`) it creates a TraceFlag record for the user which VSCode/Salesforce CLI is authenticated with. That record is then used by the org to create Apex Debug Logs for that specific user.

In your example, if User1 is the one that VSCode/Salesforce CLI is authenticated with then the Apex Debug Logs are only going to be created for User1. If you later log into that same org using User2, then you'll have to create a TraceFlag record for User2 in order for Apex Debug Logs to be created. If you do have TraceFlag records for both users then you should be able to retrieve those logs using the extensions (screenshots attached below).

**TraceFlags in the Org:**
![TraceFlag_records_](https://user-images.githubusercontent.com/1041842/56626531-e3d10f80-65f6-11e9-9412-e3eb62b588be.png)


**Available debug logs via the extensions:**
![DebugLog_records](https://user-images.githubusercontent.com/1041842/56626405-6907f480-65f6-11e9-8664-fffe489b2e42.png)


I'm also adding a link on how to create the TraceFlag records directly in the org [Set Up Debug Logging](https://help.salesforce.com/articleView?id=code_add_users_debug_log.htm&type=5)

@suneelsfdc can you confirm you're able to get debug logs using the extensions once you have TraceFlags for the users you want to debug with ?

**Comment by spelluru-indeed** [comment]
@lcampos Thanks for the detailed info. I understand that I misunderstood the concept. As per your explanation below, it doesn't seem to be creating Trace flag for Authenticated user with VSCode. Every time, it is setting trace flags for a random user(with different profiles). Could you please confirm that behavior?

**When you turn on Apex Replay Debugger (SFDX: Turn On Apex Debug Log for Replay Debugger) it creates a TraceFlag record for the user which VSCode/Salesforce CLI is authenticated with.**

**Comment by suneelsfdc** [comment]
@lcampos Thanks for the detailed info. I understand that I misunderstood the concept. As per your explanation below, it doesn't seem to be creating Trace flag for Authenticated user with VSCode. Every time, it is setting trace flags for a random user(with different profiles). Could you please confirm that behavior?

**When you turn on Apex Replay Debugger (SFDX: Turn On Apex Debug Log for Replay Debugger) it creates a TraceFlag record for the user which VSCode/Salesforce CLI is authenticated with.**

**Comment by lcampos** [comment]
Hi @suneelsfdc, I was able to replicate what you were experiencing. The root cause was that the replay debugger tries to re-use an existing TraceFlag if there is one available. In doing that we did not filter it to the specific user and that was causing the replay debugger to use a TraceFlag assigned to a different user in some cases. This is the same as reported in issue #1285 and we have a fix for it in PR #1315.
